moped-database/migrations/1651001224000_alter_project_list_view_funding/up.sql

```diff
@@ -0,0 +1,96 @@
+drop view project_list_view;
+
+CREATE VIEW public.project_list_view AS
+with project_person_list_lookup AS (
+  SELECT
+    mpp.project_id,
+    string_agg(DISTINCT concat(mu.first_name, ' ', mu.last_name, ':', mpr.project_role_name), ',') AS project_team_members
+   FROM 
+    public.moped_proj_personnel mpp
+        JOIN public.moped_users mu ON mpp.user_id = mu.user_id
+        JOIN public.moped_project_roles mpr ON mpp.role_id = mpr.project_role_id
+      WHERE (mpp.status_id = 1)
+      GROUP BY mpp.project_id
+),
+ funding_sources_lookup AS (
+  SELECT
+    mpf.project_id,
+    string_agg(mfs.funding_source_name, ', ') AS funding_source_name
+  FROM
+    moped_proj_funding mpf
+    LEFT JOIN moped_fund_sources mfs ON mpf.funding_source_id = mfs.funding_source_id
+    GROUP BY mpf.project_id
+)
```

Comment:
I added another common table expression because a normal left join was behaving similar to the team members, where if a record had more than one funding source it would make the project_features repeat.

Reply:
I think you've picked the best way to roll up the 1:many relationship here for displaying in a table with another CTE. 

I don't think that having multiple CTEs is an indicator of code smell, _unless_, the number of records returned by one is either VERY large. If that is the case, the condition that is evaluated when that CTE is joined into the query in the main `SELECT` statement has to be executed once for every row out of the CTE's set and they, by definition, are in memory without the benefits of an index. This is a common pitfall.

I think what I'm trying to say with too many words is:

Query smell comes from slowness, not length. As long as it's fast, it's probably what you want. If it's a complex query, use comments liberally and ask the database for its execution plan during development to help understand what is fast and what is slow in the query.